REFERENCE.md

```diff
@@ -200,14 +223,16 @@ Default value: ``undef``
 Data type: `String[1]`
 
 String to pass to ensure attribute for the vmwaretools x package.
+Use 'present', 'latest' to install or 'absent', 'purged' to remove package.
```

Comment:
Enum['...']

Reply:
In the old tests is test that uses explicit version numbers:


```
  context 'with custom values for parameters on machine running on vmware' do
    let(:facts) { [default_facts, { vmware_has_x: true }].reduce(:merge) }
    let(:params) do
      {
        tools_nox_package_name: 'vmware-tools-esx-nox-custom',
        tools_nox_package_ensure: '0.2-1',
        tools_x_package_name: 'vmware-tools-esx-custom',
        tools_x_package_ensure: '0.3-1',
      }
    end

    it { is_expected.to contain_package('vmware-tools-esx-nox-custom').with('ensure' => '0.2-1') }
    it { is_expected.to contain_package('vmware-tools-esx-custom').with('ensure' => '0.3-1') }
  end
```

I could add a regex that will match `[numbers].[numbers]-[numbers]` or `[numbers].[numbers].*` to also allow them.

Reply:
Fixed with `Variant[Enum['absent', 'latest', 'present', 'purged'], Pattern[/(\d+\.)+([\d-]+)/]]`.
The Regex allows strings with version numbers in the given format (eg. nn.nn-n, nn.nn.nn nn.nn nn.nn.nn-n)